scripts/build/build.sh

```diff
@@ -39,7 +39,13 @@ $REPOROOT/scripts/compile/compile.sh
 
 # Run tests on the stage2 output
 header "Testing stage2..."
+START_TOOLS=$DOTNET_TOOLS
```

Comment:
I've removed the code to save and restore PATH and just set it to stage 2 now. I think that's good enough for now since everything from this point on should be using the completed stage 2 build, and because we're running in a separate build process on all OSes, we don't have to worry about restoring the PATH.